Use UserNotifications to create/dismiss local notifications

reberhardt7 opened the pull request:
This is an extension of #50; that PR adds support for enumerating/dismissing delivered notifications
by ID, but UserNotifications notifications and UILocalNotification notifications have a different set of IDs (a notification created using UILocalNotification will have an ID that does not match the
notification's ID in the UserNotifications world and hence cannot be used to dismiss that notification). This PR creates the notifications using UserNotifications if the user is on iOS 10 so that the IDs returned by `localNotification` can be used to dismiss the notifications after they are delivered.

gran33 commented:
Thanks for a great job! 
Small question, this PR include #50 inside? Should check #50 and this together or only this one?

reberhardt7 commented:
I *think* that merging this should automatically close #50, since I merged
those commits into this branch. I'm not positive, though.

On Aug 15, 2017 12:08 AM, "Ran Greenberg" <notifications@github.com> wrote:

> Thanks for a great job!
> Small question, this PR include #50
> <https://github.com/wix/react-native-notifications/pull/50> inside?
> Should check #50
> <https://github.com/wix/react-native-notifications/pull/50> and this
> together or only this one?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wix/react-native-notifications/pull/103#issuecomment-322393297>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEoWdJ-kvIhK0WcNyFVHoN6T7AA5Tw31ks5sYUP9gaJpZM4O29m7>
> .
>

gran33 commented:
Can u plz check that all tests passed and the example project running without any errors?

reberhardt7 commented:
Yes, I will double check in the morning

On Aug 15, 2017 12:17 AM, "Ran Greenberg" <notifications@github.com> wrote:

> Can u plz check that all tests passed and the example project running
> without any errors?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wix/react-native-notifications/pull/103#issuecomment-322394662>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEoWdKkSUolaCbzT5HeKs2znFj8jCqlxks5sYUYagaJpZM4O29m7>
> .
>

reberhardt7 commented:
I rebased onto master and added a line so that categories get set properly for the new UN notifications. Tested on iOS 9 and iOS 10 and looks good to me

gran33 commented:
Thanks again for the hard work! 🤘
